src/workers/newView.ts

```diff
@@ -19,60 +22,24 @@ const addView = async (con: DataSource, entity: View): Promise<boolean> => {
     !existing ||
     entity.timestamp.getTime() - existing.timestamp.getTime() > ONE_WEEK
   ) {
-    await repo.save(entity);
-    return true;
+    const viewRecord = await repo.save(entity);
+    return { view: viewRecord, isNewView: true };
   }
-  return false;
+  return { view: existing, isNewView: false };
 };
 
-const DEFAULT_TIMEZONE = 'UTC'; // in case user doesn't have a timezone set
-const INC_STREAK_QUERY = `
```

Comment:
@kopancek can you take a look at this changes since you were the author of original query. I am all up for removing it since typeorm logic is much easier to parse mentally but I am not 100% if there is some edge case this new implementation does not cover. 

I see some tests fail so something might be off.

Reply:
looking into specs now, they may be failing due to how they are set up, but yes, would love @kopancek eyes on this as well 🙏